### PR TITLE
1612023 brand

### PR DESCRIPTION
**Pull request by Erul98**
Please help me review my code. Thanks!

### CODE REVIEW

File: services/backend-service/src/modules/googledrive/googledrive.service.ts [hunk 1]
```diff
@@ -0,0 +1,187 @@
...
+  async uploadFile(payload: Partial<UploadDTO>): Promise<UploadStatusResponse> {
+    upload_response_json.reNewData();
+    let isFileExist = false;
+    let folder_id = '';
+    try {
+      // Create the folder with the name is user id if the file the first upload file
+      const folders = await drive.files.list({
+        q: `mimeType = 'application/vnd.google-apps.folder' and trashed=false`,
+      });
+      for (let i = 0; i < folders.data.files.length; i++) {
```
Comment (buiminhhai1) [inline services/backend-service/src/modules/googledrive/googledrive.service.ts:43]: use forEach instead

File: services/backend-service/src/modules/googledrive/googledrive.service.ts [hunk 2]
```diff
@@ -0,0 +1,187 @@
...
+        if (folders.data.files[i].name == this.context.userId) {
+          folder_id = folders.data.files[i].id;
```
Comment (buiminhhai1) [inline services/backend-service/src/modules/googledrive/googledrive.service.ts:45]: camel naming please

File: services/backend-service/src/modules/googledrive/googledrive.service.ts [hunk 3]
```diff
@@ -0,0 +1,187 @@
...
+          isFileExist = true;
+          break;
+        }
+      }
+      // Check folder name is exist
+      if (!isFileExist) {
+        const filemetadata = {
+          name: this.context.userId,
+          mimeType: 'application/vnd.google-apps.folder',
```
Comment (buiminhhai1) [inline services/backend-service/src/modules/googledrive/googledrive.service.ts:54]: define constant for this

File: services/backend-service/src/modules/googledrive/googledrive.service.ts [hunk 4]
```diff
@@ -0,0 +1,187 @@
...
+        };
+        const new_folder = await drive.files.create({
```
Comment (buiminhhai1) [inline services/backend-service/src/modules/googledrive/googledrive.service.ts:56]: camel naming

File: services/backend-service/src/modules/googledrive/googledrive.service.ts [hunk 5]
```diff
@@ -0,0 +1,187 @@
...
+        mimeType: payload.type,
+        body: bufferStream,
+      };
+      // waiting for create file on drive
+      const response = await drive.files.create({
+        requestBody: fileMetaData,
+        media: media,
+      });
+      //
+      upload_response_json.statusCode = response.status;
```
Comment (buiminhhai1) [inline services/backend-service/src/modules/googledrive/googledrive.service.ts:80]: don't need handle status code

File: services/backend-service/src/modules/googledrive/googledrive.service.ts [hunk 6]
```diff
@@ -0,0 +1,187 @@
...
+          name: response.data.name,
+          type: response.data.mimeType,
+        };
+        return upload_response_json;
+      } else {
+        return upload_response_json;
+      }
+    } catch (e) {
+      //console.log(e);
+      upload_response_json.statusCode = 500;
```
Comment (buiminhhai1) [inline services/backend-service/src/modules/googledrive/googledrive.service.ts:93]: just use internalexception

File: services/backend-service/src/modules/googledrive/googledrive.service.ts [hunk 7]
```diff
@@ -0,0 +1,187 @@
...
+      return upload_response_json;
```
Comment (buiminhhai1) [inline services/backend-service/src/modules/googledrive/googledrive.service.ts:94]: camel naming

File: services/backend-service/src/modules/googledrive/googledrive.service.ts [hunk 8]
```diff
@@ -0,0 +1,187 @@
...
+    }
+  }
+  async uploadDb(folderId: string): Promise<boolean> {
+    const google_drive_model = await this.googleDriveRepo
+      .createQueryBuilder('google_drive')
+      .where('google_drive.userId = :userId', { userId: this.context.userId })
+      .getOne();
+    if (google_drive_model == null) {
```
Comment (buiminhhai1) [inline services/backend-service/src/modules/googledrive/googledrive.service.ts:102]: camel naming, can use is lodash isEmpty

File: services/backend-service/src/modules/googledrive/googledrive.service.ts [hunk 9]
```diff
@@ -0,0 +1,187 @@
...
+      await this.googleDriveRepo.save(
+        this.googleDriveRepo.create({ folderId: folderId, user: { id: this.context.userId } }),
+      );
+    } else {
+      google_drive_model.folderId = folderId;
```
Comment (buiminhhai1) [inline services/backend-service/src/modules/googledrive/googledrive.service.ts:107]: naming

File: services/backend-service/src/modules/googledrive/googledrive.service.ts [hunk 10]
```diff
@@ -0,0 +1,187 @@
...
+        .getOne();
+      if (google_drive_model != null) {
+        const files = await drive.files.list({
+          q: `'${google_drive_model.folderId}' in parents and trashed=false`,
+        });
+        const arrayList = [];
+        for (let i = 0; i < files.data.files.length; i++) {
+          arrayList.push({
+            fildeId: files.data.files[i].id,
+            ...(await this.generatePublicUrl(files.data.files[i].id)),
```
Comment (buiminhhai1) [inline services/backend-service/src/modules/googledrive/googledrive.service.ts:127]: using promise all

File: services/backend-service/src/modules/googledrive/googledrive.service.ts [hunk 11]
```diff
@@ -0,0 +1,187 @@
...
+      const google_drive_model = await this.googleDriveRepo
+        .createQueryBuilder('google_drive')
+        .where('google_drive.userId = :userId', { userId: this.context.userId })
+        .getOne();
+      if (google_drive_model != null) {
+        const files = await drive.files.list({
+          q: `'${google_drive_model.folderId}' in parents and trashed=false`,
+        });
+        const arrayList = [];
+        for (let i = 0; i < files.data.files.length; i++) {
```
Comment (buiminhhai1) [inline services/backend-service/src/modules/googledrive/googledrive.service.ts:124]: forEach

File: services/backend-service/src/modules/googledrive/googledrive.service.ts [hunk 12]
```diff
@@ -0,0 +1,187 @@
...
+          arrayList.push({
+            fildeId: files.data.files[i].id,
+            ...(await this.generatePublicUrl(files.data.files[i].id)),
+          });
+        }
+
+        return {
+          statusCode: files.status,
```
Comment (buiminhhai1) [inline services/backend-service/src/modules/googledrive/googledrive.service.ts:132]: dont need using status code

File: services/backend-service/src/modules/googledrive/googledrive.service.ts [hunk 13]
```diff
@@ -0,0 +1,187 @@
...
+          data: arrayList,
+        };
+      } else {
+        return {
+          statusCode: 400,
+          data: null,
+        };
+      }
+    } catch (error) {
```
Comment (buiminhhai1) [inline services/backend-service/src/modules/googledrive/googledrive.service.ts:141]: remove unnecessary try catch

File: services/backend-service/src/modules/googledrive/googledrive.service.ts [hunk 14]
```diff
@@ -0,0 +1,187 @@
...
+      const result = await drive.files.get({
+        fileId: fileId,
+        fields: 'webViewLink, webContentLink',
+      });
+      return {
+        link: result.data.webViewLink,
+        download: result.data.webContentLink,
+      };
+    } catch (error) {
+      return {
```
Comment (buiminhhai1) [inline services/backend-service/src/modules/googledrive/googledrive.service.ts:168]: throw new InternalException

File: services/backend-service/src/modules/googledrive/googledrive.service.ts [hunk 15]
```diff
@@ -0,0 +1,187 @@
...
+        link: '',
+        download: '',
+      };
+    }
+  }
+
+  async deleteFile(_query: Partial<DeleteDTO>): Promise<DeleteDataResponse> {
+    try {
```
Comment (buiminhhai1) [inline services/backend-service/src/modules/googledrive/googledrive.service.ts:176]: remove try catch and naming

File: services/backend-service/src/modules/googledrive/googledrive.service.ts [hunk 16]
```diff
@@ -0,0 +1,187 @@
...
+      //console.log(_query);
+      const response = await drive.files.delete({
+        fileId: _query.fileId,
+      });
+      return { statusCode: response.status, data: response.data };
+    } catch (error) {
+      //console.log(error);
+      return { statusCode: 500, data: null };
```
Comment (buiminhhai1) [inline services/backend-service/src/modules/googledrive/googledrive.service.ts:184]: remove status code

**Comment by Erul98** [comment]
Please help me review

**Pull request closed without merging**